### PR TITLE
Exporting missing instructors as CSV

### PR DESCRIPTION
**Pull request by gvwilson**
The certificate generator needs a CSV file as input, and the award dates, so...

### CODE REVIEW

File: workshops/management/commands/check_certificates.py [hunk 1]
```diff
@@ -46,13 +50,13 @@ def get_file_people(self, path):
...
-                try:
-                    p = Person.objects.get(username=i)
-                    print(' {0}: {1}'.format(i, p))
-                except Person.DoesNotExist:
-                    print(' {0}'.format(i))
+    def missing(self, report, title, kind, usernames, records):
+        '''Report missing usernames.'''
+        for u in usernames:
+            try:
+                p = Person.objects.get(username=u)
```
Comment (pbanaszkiewicz) [inline workshops/management/commands/check_certificates.py]: Minor: I'd rather see `for username in usernames`.

File: workshops/management/commands/check_certificates.py [hunk 2]
```diff
@@ -19,15 +20,18 @@ def handle(self, *args, **options):
...
             cert_path = os.path.join(path_to_root, name)
             if not os.path.isdir(cert_path):
-                print('No directory {0}'.format(name))
+                print('No directory {0}'.format(name), file=sys.stderr)
             else:
                 file_people = self.get_file_people(cert_path)
-                self.report_missing('database but not disk', name, db_people - file_people)
-                self.report_missing('disk but not database', name, file_people - db_people)
+                self.missing(result, 'database-disk', name, db_people - file_people, db_records)
+                self.missing(result, 'disk-database', name, file_people - db_people, db_records)
```
Comment (pbanaszkiewicz) [inline workshops/management/commands/check_certificates.py]: Does this work? Can you really do `dict() - set()` or `set() - dict()`?

Comment (gvwilson) [inline workshops/management/commands/check_certificates.py]: `db_people` is a set created with `db_records.keys()`

On 2016-02-24 10:11 AM, Piotr Banaszkiewicz wrote:
>
> In workshops/management/commands/check_certificates.py 
> <https://github.com/swcarpentry/amy/pull/708#discussion_r53951759>:
>
> >              else:
> >                  file_people = self.get_file_people(cert_path)
> > -                self.report_missing('database but not disk', name, db_people - file_people)
> > -                self.report_missing('disk but not database', name, file_people - db_people)
> > +                self.missing(result, 'database-disk', name, db_people - file_people, db_records)
> > +                self.missing(result, 'disk-database', name, file_people - db_people, db_records)
>
> Does this work? Can you really do |dict() - set()| or |set() - dict()|?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/swcarpentry/amy/pull/708/files#r53951759>.
>

-- 
Dr Greg Wilson
Director of Instructor Training
Software Carpentry Foundation

File: workshops/management/commands/check_certificates.py (hunk 1 above)
Comment (gvwilson) [inline workshops/management/commands/check_certificates.py]: I find that long variable names that different only in the last character are hard to read, and that short names are OK for iteration variables.

**Pull request merged**